hashicorp/vault
# Add Browserstack for IE11 testing

# Add Browserstack for IE11 Testing

This PR implements Browserstack & [ember-cli-browserstack](https://github.com/kategengler/ember-cli-browserstack) for automated cross-browser testing. At the moment we are only testing the Open Source secrets related acceptance tests in IE11 on Windows 10 due to problems when running the test suite concurrently on multiple browsers. In the future we can easily swap out IE11 for a different browser. 

## Summary of changes
- Refactored lots of test code to be compatible with IE11 (using `objectAt`, `Blob` instead of `File` etc)
- Added a new CI build job run the tests in Browserstack against each PR

## Testing Instructions
1. Log in to [Browserstack](https://automate.browserstack.com/) to get your username and access key. 
2. Set your `BROWSERSTACK_USERNAME` and `BROWSERSTACK_ACCESS_KEY` as environment variables
3. Run `yarn run test:browserstack`
- [ ] Verify you can see the tests running at https://localhost:7357/
- [ ] Verify you can see the tests running on [Browserstack](https://automate.browserstack.com/)
